x-pack/plugins/ml/public/application/datavisualizer/data_comparison/data_drift_index_patterns_editor.tsx

```diff
@@ -0,0 +1,370 @@
+              />
+            }
+            dataViewEditorService={referenceDataViewEditorService}
+            indexPattern={referenceIndexPattern}
+            setIndexPattern={setReferenceIndexPattern}
+          />
+        </EuiFlexItem>
+      ),
+    },
+    {
+      title: 'Pick index pattern for comparison data indices',
+      children: (
+        <EuiFlexItem grow={false}>
+          <DataViewEditor
+            key={'comparison'}
+            label={
+              <FormattedMessage
+                id="xpack.ml.dataDrift.indexPatternsEditor.comparisonData"
+                defaultMessage="Index pattern for comparison data indices"
+              />
+            }
+            dataViewEditorService={productionDataViewEditorService}
+            indexPattern={productionIndexPattern}
+            setIndexPattern={setProductionIndexPattern}
+          />
+        </EuiFlexItem>
+      ),
+    },
+    {
+      title: 'Additional settings',
+      children: (
+        <EuiFlexGroup direction="column">
+          {combinedTimeFieldOptions.length > 0 ? (
+            <EuiFormRow
+              label={i18n.translate('xpack.ml.dataDrift.indexPatternsEditor.timeField', {
+                defaultMessage: 'Time field',
+              })}
+              fullWidth
+              color={'disabled'}
+            >
+              <>
+                <EuiComboBox<string>
+                  placeholder={i18n.translate(
+                    'xpack.ml.dataDrift.indexPatternsEditor.timestampField',
+                    {
+                      defaultMessage: 'Select an optional timestamp field',
+                    }
+                  )}
+                  singleSelection={{ asPlainText: true }}
+                  options={combinedTimeFieldOptions}
+                  selectedOptions={timeField}
+                  onChange={(newValue) => {
+                    if (newValue.length === 0) {
+                      // Don't allow clearing the type. One must always be selected
+                      return;
+                    }
+                    setTimeField(newValue);
+                  }}
+                  isClearable={false}
+                  isDisabled={productionIndexPattern === '' && referenceIndexPattern === ''}
+                  data-test-subj="timestampField"
+                  aria-label={i18n.translate(
+                    'xpack.ml.dataDrift.indexPatternsEditor.timestampSelectAriaLabel',
+                    {
+                      defaultMessage: 'Timestamp field',
+                    }
+                  )}
+                  fullWidth
+                />
+              </>
+            </EuiFormRow>
+          ) : null}
+          {!foundDataViewId ? (
+            <EuiFormRow
+              label={i18n.translate('xpack.ml.dataDrift.indexPatternsEditor.dataViewName', {
+                defaultMessage: 'Data view name',
+              })}
+              helpText={`Optional data view name. ${
```

Comment:
It looks like it is appending the time field name here to the list of index patterns? Is that intentional? I'm not sure you need to append the timestamp field too.

<img width="477" alt="image" src="https://github.com/elastic/kibana/assets/7405507/6cf6cb47-68d1-4189-8809-62e5f6ca62a8">

Reply:
This should only happen if there is already a data view with the same index patterns, but different time field. Otherwise, there will be a conflict in the name.